CMakeLists.txt

```diff
@@ -62,6 +62,15 @@ option(SIRF_INSTALL_DEPENDENCIES "Install dlls etc" WIN32)
 ####### CMake path
 set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
+# Define the pre-commit hook for developer
+find_package(Git)
+if(GIT_FOUND)
+  message(STATUS "Found Git")
+  file(COPY "${CMAKE_SOURCE_DIR}/update_copyright_hook" DESTINATION "${CMAKE_SOURCE_DIR}/.git/hooks" USE_SOURCE_PERMISSIONS)
+  file(RENAME "${CMAKE_SOURCE_DIR}/.git/hooks/update_copyright_hook" "${CMAKE_SOURCE_DIR}/.git/hooks/pre-commit")
+  execute_process(COMMAND chmod +x "${CMAKE_SOURCE_DIR}/.git/hooks/pre-commit")
```

Comment:
does CMake allow changing permissions? This wouldn't work on Windows of course. In fact, neither would the hook  itself, so let's put the whole thing in `if NOT Windows`. Of course, that'd mean we leave out @evgueni-ovtchinnikov who puts in 90% of the commits...

Reply:
> does CMake allow changing permissions?

Changing permission works on my machine (and is required for the hook to run).

> This wouldn't work on Windows of course.

From this link, I think it **might** work on Windows: https://www.tygertec.com/git-hooks-practical-uses-windows/. Maybe it uses `git bash` on windows to be able to execute? Not sure.

Reply:
The link also says that `chmod +x` on windows just checks NTFS execution rights. Again, no idea but should be easy enough to test.

Reply:
hmm also this may overwrite people's own custom `.git/hooks/pre-commit`